packages/components/src/menu/stories/index.story.tsx

```diff
@@ -64,300 +64,337 @@ const meta: Meta< typeof Menu > = {
+						</Menu.ItemHelpText>
+					</Menu.Item>
+					<Menu.Item hideOnClick={ false }>
+						<Menu.ItemLabel>Label</Menu.ItemLabel>
+						<Menu.ItemHelpText>
+							This item doesn&apos;t close the menu on click
+						</Menu.ItemHelpText>
 					</Menu.Item>
+					<Menu.Item disabled>Disabled item</Menu.Item>
+					<Menu.Separator />
+					<Menu.Group>
+						<Menu.GroupLabel>Group label</Menu.GroupLabel>
+						<Menu.Item
+							prefix={ <Icon icon={ customLink } size={ 24 } /> }
+						>
+							<Menu.ItemLabel>With prefix</Menu.ItemLabel>
+						</Menu.Item>
+						<Menu.Item suffix="⌘S">With suffix</Menu.Item>
+						<Menu.Item
+							disabled
+							prefix={
+								<Icon icon={ formatCapitalize } size={ 24 } />
+							}
+							suffix="⌥⌘T"
+						>
+							<Menu.ItemLabel>
+								Disabled with prefix and suffix
+							</Menu.ItemLabel>
+							<Menu.ItemHelpText>And help text</Menu.ItemHelpText>
+						</Menu.Item>
+					</Menu.Group>
+				</Menu.Popover>
+			</>
+		),
+	},
+};
+
+export const WithSubmenu: Story = {
+	args: {
+		children: (
+			<>
+				<Menu.TriggerButton
+					render={
+						<Button __next40pxDefaultSize variant="secondary" />
+					}
+				>
+					Open menu
+				</Menu.TriggerButton>
+				<Menu.Popover>
+					<Menu.Item>Level 1 item</Menu.Item>
 					<Menu>
-						<Menu.SubmenuTriggerItem>
-							<Menu.ItemLabel>Submenu trigger</Menu.ItemLabel>
+						<Menu.SubmenuTriggerItem suffix="Suffix">
+							<Menu.ItemLabel>
+								Submenu trigger item with a long label
+							</Menu.ItemLabel>
 						</Menu.SubmenuTriggerItem>
 						<Menu.Popover>
 							<Menu.Item>
-								<Menu.ItemLabel>Level 3 item</Menu.ItemLabel>
+								<Menu.ItemLabel>Level 2 item</Menu.ItemLabel>
 							</Menu.Item>
 							<Menu.Item>
-								<Menu.ItemLabel>Level 3 item</Menu.ItemLabel>
+								<Menu.ItemLabel>Level 2 item</Menu.ItemLabel>
 							</Menu.Item>
+							<Menu>
+								<Menu.SubmenuTriggerItem>
+									<Menu.ItemLabel>
+										Submenu trigger
+									</Menu.ItemLabel>
+								</Menu.SubmenuTriggerItem>
+								<Menu.Popover>
+									<Menu.Item>
+										<Menu.ItemLabel>
+											Level 3 item
+										</Menu.ItemLabel>
+									</Menu.Item>
+									<Menu.Item>
+										<Menu.ItemLabel>
+											Level 3 item
+										</Menu.ItemLabel>
+									</Menu.Item>
+								</Menu.Popover>
+							</Menu>
 						</Menu.Popover>
 					</Menu>
 				</Menu.Popover>
-			</Menu>
-		</Menu.Popover>
-	</Menu>
-);
-WithSubmenu.args = {
-	...Default.args,
+			</>
+		),
+	},
 };
 
-export const WithCheckboxes: StoryFn< typeof Menu > = ( props: MenuProps ) => {
-	const [ isAChecked, setAChecked ] = useState( false );
-	const [ isBChecked, setBChecked ] = useState( true );
-	const [ multipleCheckboxesValue, setMultipleCheckboxesValue ] = useState<
-		string[]
-	>( [ 'b' ] );
+export const WithCheckboxes: Story = {
+	render: function WithCheckboxes( props: MenuProps ) {
```

Comment:
These parameter types shouldn't be necessary, since that is already handled through the `StoryObj<>` generic.

```suggestion
	render: function WithCheckboxes( props ) {
```

Reply:
I did that initially, but then I added back the explicit props because it seems to improve TypeScript performance (TS doesn't need to compute the prop types since they are explicitly declared)

Reply:
We ran some TS profiling here, and it turned out to make a big difference. Results were similar to here:

https://github.com/WordPress/gutenberg/pull/67422#discussion_r1883524031

As I discussed with @ciampo over DM earlier, I think this has something to do with how Storybook infers component props for stories when working with compound components, and maybe the inference is not optimized to work with subcomponents and does extra cycles, resulting in infinite loop. It's worth debugging with a dev Storybook setup and an isolated story test case, IMO.

Reply:
Huh, that's unfortunate!

One ideological issue I have with `props: MenuProps` is that `MyComponentProps` is often not the canonically accurate set of props in our library, because we often mix in element props with `WordPressComponentProps<>`. I feel like we should generally prefer `React.ComponentProps< typeof MyComponent >` for accuracy. Would that still work here while maintaining the performance benefits? (Not a blocker though.)

Reply:
Let's try it out and measure it, but I think it should be fine, as long as we don't let Storybook infer the prop types.

Reply:
What about `React.ComponentProps< Menu >` ? It could hit a sweet spot between accuracy and performance?